### PR TITLE
fix(anta.tests): Cleaning up Security tests module (VerifySpecificIPSecConn)

### PR DESCRIPTION
**Pull request by vitthalmagadum**
# Description

Refactoring Security tests module to address the following issues:

- #809
- #587
- #388

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have run pre-commit for code linting and typing (`pre-commit run`)
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: anta/tests/security.py [hunk 1]
```diff
@@ -719,35 +727,16 @@ class VerifySpecificIPSecConn(AntaTest):
...
-            """Optional list of IPv4 security connections of a peer."""
-
-            class IPSecConn(BaseModel):
-                """Details of IPv4 security connections for a peer."""
-
-                source_address: IPv4Address
-                """Source IPv4 address of the connection."""
-                destination_address: IPv4Address
-                """Destination IPv4 address of the connection."""
+        IPSecPeers: ClassVar[type[IPSecPeers]] = IPSecPeers
```
Comment (gmuloc) [inline anta/tests/security.py:739]: Add a comment that it is for backward compatibility on this one as it feels weird to have it not used for anyting

Comment (vitthalmagadum) [reply anta/tests/security.py:738]: updated

File: anta/tests/security.py [hunk 2]
```diff
@@ -692,15 +692,23 @@ def test(self) -> None:
 
 
 class VerifySpecificIPSecConn(AntaTest):
-    """Verifies the state of IPv4 security connections for a specified peer.
+    """Verifies the IPv4 security connections.
 
-    It optionally allows for the verification of a specific path for a peer by providing source and destination addresses.
-    If these addresses are not provided, it will verify all paths for the specified peer.
+    This test performs the following checks for each specified address family:
```
Comment (gmuloc) [inline anta/tests/security.py:697]:
```suggestion
    This test performs the following checks for each peer:
```

File: anta/tests/security.py [hunk 3]
```diff
@@ -692,15 +692,23 @@ def test(self) -> None:
...
+
+      1. Validates that the VRF is configured.
+      2. Checks for the presence of IPv4 security connections for the specified peer.
+      3. For each relevant peer:
+        - If source and destination addresses are provided, Verifies the security connection for the specific path.
```
Comment (gmuloc) [inline anta/tests/security.py:702]:
```suggestion
        - If source and destination addresses are provided, verifies the security connection for the specific path exists and is `Established`.
```

File: anta/tests/security.py [hunk 4]
```diff
@@ -692,15 +692,23 @@ def test(self) -> None:
...
+        - If no addresses are provided, Verifies all security connections associated with the peer.
```
Comment (gmuloc) [inline anta/tests/security.py:703]:
```suggestion
        - If no addresses are provided, verifies that all security connections associated with the peer are `Established`.
```

File: anta/tests/security.py [hunk 5]
```diff
@@ -692,15 +692,23 @@ def test(self) -> None:
...
+        - Verifies that the connection is in the `Established` state.
```
Comment (gmuloc) [inline anta/tests/security.py:704]:
```suggestion
```

File: anta/tests/security.py [hunk 6]
```diff
@@ -692,15 +692,23 @@ def test(self) -> None:
...
 
     Expected Results
     ----------------
-    * Success: The test passes if the IPv4 security connection for a peer is established in the specified VRF.
-    * Failure: The test fails if IPv4 security is not configured, a connection is not found for a peer, or the connection is not established in the specified VRF.
+    * Success: If all checks pass for all specified IPv4 security connections.
+    * Failure: If any of the following occur:
+        - No IPv4 security connections are found for the peer
+        - The security connection is not established for the specified path or any of the peer's connections when no path is specified.
```
Comment (gmuloc) [inline anta/tests/security.py:711]:
```suggestion
        - The security connection is not established for the specified path or any of the peer connections is not established when no path is specified.
```

**Review by gmuloc — APPROVED** [review]

**Pull request merged**